### PR TITLE
Using kpm list in KRuntime's makefile

### PR DESCRIPTION
**Pull request by troydai**
Use kpm list to produce a smaller package.

### CODE REVIEW

**Comment by troydai** [comment]
@davidfowl

File: makefile.shade [hunk 1]
```diff
@@ -725,11 +650,89 @@ var RC_FILES = '${FindAllFiles("Resource.rc", "dotnet", "dotnet.net45", "dotnet.
 
     exec program='cmd' commandline='/C dotnetsdk alias build ${matchVersion}' if='!IsMono && matchVersion != ""' 
     exec program='cmd' commandline='/C start cmd /K dotnetsdk use build' if='!IsMono && matchVersion != ""' 
+
+#update-tpa
+    @{
+        var envKreTrace = Environment.GetEnvironmentVariable("KRE_TRACE");
```
Comment (davidfowl) [inline makefile.shade]: DOTNET_TRACE

File: makefile.shade [hunk 2]
```diff
@@ -725,11 +650,89 @@ var RC_FILES = '${FindAllFiles("Resource.rc", "dotnet", "dotnet.net45", "dotnet.
...
+    int ExecuteAndRedirectOutput(string command, string argument, string outputFile)
+    {
+        var procStartInfo = new ProcessStartInfo 
+        {
+            FileName = command, 
+            Arguments = argument, 
+            WorkingDirectory = Directory.GetCurrentDirectory(),
+            UseShellExecute = false
+        };
+        procStartInfo.EnvironmentVariables["KRE_TRACE"] = "0";
```
Comment (davidfowl) [inline makefile.shade]: DOTNET_TRACE

File: makefile.shade (hunk 1 above)
Comment (troydai) [inline makefile.shade]: got it.

File: makefile.shade (hunk 2 above)
Comment (troydai) [inline makefile.shade]: aye aye

File: makefile.shade [hunk 3]
```diff
@@ -725,11 +650,89 @@ var RC_FILES = '${FindAllFiles("Resource.rc", "dotnet", "dotnet.net45", "dotnet.
...
+            Exec("cmd", string.Format(@"/C kpm list {0} --assemblies --runtime {1} > {2}",
+                Path.Combine(ROOT, "src", "dotnet.core45.managed"),
+                Path.Combine(CORECLR_TARGET_PATH, "Runtime", "x86"),
+                dotnetcore45DependencyList ));
+
+            UpdateTpa(dotnetcore45DependencyList , tpacppFile);
+        }
+        finally
+        {
+            Environment.SetEnvironmentVariable("KRE_TRACE", envKreTrace);
```
Comment (davidfowl) [inline makefile.shade]: DOTNET_TRACE

File: makefile.shade [hunk 4]
```diff
@@ -725,11 +650,89 @@ var RC_FILES = '${FindAllFiles("Resource.rc", "dotnet", "dotnet.net45", "dotnet.
...
+
+#update-tpa
+    @{
+        var envKreTrace = Environment.GetEnvironmentVariable("KRE_TRACE");
+
+        try
+        {
+            var dotnetcore45DependencyList = Path.Combine(BUILD_DIR2, "dotnetcore45-dependencies.txt");
+            var tpacppFile = Path.Combine(ROOT, "src", "dotnet.core45", "tpa.cpp");
+            Environment.SetEnvironmentVariable("KRE_TRACE", "0");
```
Comment (victorhurdugaci) [inline makefile.shade]: DOTNET_TRACE

**Comment by troydai** [comment]
This is the end of KRE_TRACE hunting.

**Comment by troydai** [comment]
Rebased on dev.

**Comment by troydai** [comment]
So many things were changed. I'll update and send another PR.

**Pull request closed without merging**